gorm/transaction.go

```diff
@@ -57,15 +81,29 @@ func (t *Transaction) AddAfterCommitHook(hooks ...func(context.Context)) {
 	t.afterCommitHook = append(t.afterCommitHook, hooks...)
 }
 
-// BeginFromContext will extract transaction wrapper from context and start new transaction.
+// GetReadOnlyDBInstance returns the read only database instance stored in the ctx
+func GetReadOnlyDBInstance(ctx context.Context) (*gorm.DB, error) {
```

Comment:
Let's not export this unless there's a reason to? `getReadOnlyDBInstance`

With `GetReadOnlyDB` and `BeginFromContext` tested, I don't think there's much value in specifically testing this function, so I don't think a unit test for it alone is required.